sdk-lib/src/main/java/cash/z/ecc/android/sdk/block/processor/CompactBlockProcessor.kt

```diff
@@ -1428,6 +1461,15 @@ class CompactBlockProcessor internal constructor(
                     }.buffer(1).map { downloadStageResult ->
                         Twig.debug { "Download stage done with result: $downloadStageResult" }
 
+                        // Triggering UTXOs fetch operation
+                        emit(
+                            BatchSyncProgress(
+                                order = downloadStageResult.batch.order,
+                                range = downloadStageResult.batch.range,
+                                resultState = SyncingResult.FetchUtxos
+                            )
+                        )
```

Comment:
That's why we've implemented this follow-up issue to run the fetchUtxos action every 1000 instead of every 100 blocks. The changes in this PR follow the connected issue request:

> When I added transparent UTXO fetching to zec-sqlite-cli in https://github.com/Electric-Coin-Company/zec-sqlite-cli/pull/23, I instead implemented it by, just after calling update_chain_tip and before any shielded scanning in a scan cycle, fetching all UTXOs starting at the block-fully-scanned height (which is always increasing, and means we fetch the full block range where we have not scanned all shielded information).

Does that mean you call it just once and not periodically?

Reply:
In `zec-sqlite-cli` I call it once per `updateChainTip`. So it is periodic but at a much lower frequency, and depending on the structure of the wallet (specifically at what points it receives notes) there could be extended gaps between calls.

I expect that the "right" frequency will be somewhere in between, or perhaps a hybrid (once per `updateChainTip` or per X blocks, whichever occurs sooner since the last time it was called).

Reply:
Filed a new fetch UTXOs frequency improving issue https://github.com/Electric-Coin-Company/zcash-android-wallet-sdk/issues/1552